From 52ed3fb5981cddcd84d787912fa548043ff568da Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 09 Jun 2010 16:29:08 +0000
Subject: Fix thunderbird's mailto command

Seeing as it's the only mailer to be such a pain.

https://bugzilla.gnome.org/show_bug.cgi?id=614222
---
diff --git a/src/plugins/evolution/evolution.c b/src/plugins/evolution/evolution.c
index a25ade1..8e1dd0c 100644
--- a/src/plugins/evolution/evolution.c
+++ b/src/plugins/evolution/evolution.c
@@ -99,9 +99,18 @@ init (NstPlugin *plugin)
 		/* Find what the default mailer is */
 		if (strstr (mail_cmd, "balsa"))
 			type = MAILER_BALSA;
-		else if (strstr (mail_cmd, "thunder") || strstr (mail_cmd, "seamonkey"))
+		else if (strstr (mail_cmd, "thunder") || strstr (mail_cmd, "seamonkey")) {
+			char **strv;
+
 			type = MAILER_THUNDERBIRD;
-		else if (strstr (mail_cmd, "sylpheed") || strstr (mail_cmd, "claws"))
+
+			/* Thunderbird sucks, see
+			 * https://bugzilla.gnome.org/show_bug.cgi?id=614222 */
+			strv = g_strsplit (mail_cmd, " ", -1);
+			g_free (mail_cmd);
+			mail_cmd = g_strdup_printf ("%s %%s", strv[0]);
+			g_strfreev (strv);
+		} else if (strstr (mail_cmd, "sylpheed") || strstr (mail_cmd, "claws"))
 			type = MAILER_SYLPHEED;
 		else if (strstr (mail_cmd, "anjal"))
 			type = MAILER_EVO;
--
cgit v0.8.3.1
